pkg/apis/autopilot/v1alpha1/autopilotrule.go

```diff
@@ -100,6 +99,50 @@ type AutopilotRuleSpec struct {
 	ActionsCoolDownPeriod int64 `json:"actionsCoolDownPeriod,omitempty"`
 }
 
+// AutopilotRuleStatus captures the status of various objects that are monitored as part of an autopilot rule
+type AutopilotRuleStatus struct {
+	// Objects is a map of objects that are monitored as part of a autopilot rule and their statuses
+	Objects map[RuleStatusObjectKey][]*AutopilotRuleObjectStatus `json:"objects"`
+}
+
+// AutopilotRuleObjectStatus represents status of a particular object that is being monitored by autopilot
+type AutopilotRuleObjectStatus struct {
+	// Name of the object
+	Name string `json:"name"`
+	// LastProcessTimestamp was the last time the object was processed
+	LastProcessTimestamp meta.Time `json:"lastProcessTimestamp"`
+	// State of the object
+	State RuleState `json:"state"`
+	// Message is the user friendly status
+	Message string `json:"message"`
+	// TODO add NextProcessTimestamp
+}
+
+// RuleState is the type for the state of a rule
+type RuleState string
+
+const (
+	// RuleStateInit is the initial state of the rule where monitorign has not yet begin
+	RuleStateInit RuleState = "Initializing"
+	// RuleStateNormal is when the rule is being monitored and is in normal state
+	RuleStateNormal RuleState = "Normal"
+	// RuleStateTriggered is when the rule has it's conditions met
+	RuleStateTriggered RuleState = "Triggered"
+	// RuleStateActivePending is when the rule has it's conditions met but the actions are
+	// not being performed yet.
+	RuleStateActivePending RuleState = "ActivePending"
```

Comment:
ActivePending -> ActiveRulePending? (again, user can be confused since there are actions and rules can appear)

Reply:
These constants were just moved from the autopilot repo. Previously we had avoided adding actions and rules in the constant names as they seemed redundant.

But your observation is correct. Being verbose is a better route to go as that will bring clarity.

Reply:
renamed to ActiveActionsPending